### PR TITLE
Kubernetes 1.6 support.

### PR DESCRIPTION
**Pull request by izaakschroeder**
Adds a few handy features:

 * Token authentication.
 * In-Cluster server creation.
 * Support for `insecure-skip-tls-verify` in `kubeconfig`.
 * Upgrade spec to K8S 1.6.

Unfortunately the 1.6 spec changes the module namespace quite a bit so this is definitely a major version bump.

I have one existing test to fix up but otherwise everything else is working well. 👍 I should probably add tests for the new features yet too.

**NOTE**: I will rebase this when it's ready for merging.

### CODE REVIEW

File: lib/kazan/models.ex [hunk 1]
```diff
@@ -39,7 +39,18 @@ defmodule Kazan.Models do
     cond do
       kind -> {:ok, kind}
       Map.has_key?(data, "kind") and Map.has_key?(data, "apiVersion") ->
-        version = data["apiVersion"] |> String.split("/") |> List.last
+        version = case data["apiVersion"] do
+          "v" <> v ->
```
Comment (izaakschroeder) [inline lib/kazan/models.ex]: TODO: This logic also needs to support the `Kazan.Models.Io.K8s.Apimachinery.Pkg.Apis.Meta.V1` namespace. Not sure what the `apiVersion` is for these objects.

**Comment by obmarg** [comment]
Nice work @izaakschroeder - thanks for the contribution. 

I'm a little busy just now, but I'll try and give this a proper look over soon.

**Comment by izaakschroeder** [comment]
Thanks @obmarg! I know that busy feeling – no rush. 😄

**Comment by obmarg** [comment]
Hey @izaakschroeder - really sorry it's taken me so long to get back to this PR.  

I've taken a few of your commits and merged them onto master separately from this.

Hoping to get the rest merged soon - just need to figure out what to do about your TODO in the comment above, and make a decision around how to namespace the models [as described in this comment here](https://github.com/obmarg/kazan/pull/4#issuecomment-326754089).  Hopefully shouldn't take too long.

**Comment by obmarg** [comment]
I've finally got all of this work merged, so I'm going to close this PR.  I've made a few tweaks to the way we build model module names as well, so things aren't quite how this PR looked.  Tested against a kube 1.6 cluster and it seems to be working though.

The work has been added to master in 0f214c1600d0dab5dca1131fc0fce8a508765a38, 4ac48d66ee25241ac48ccb44939f74c2b93c109a, a17193187cc6f8def7cc8389acff0f0cb0ce187b, 0ba1cce145998dbd94fde191b3614fad2016992a and 1f71f823549b4b7790bf1a26874513ad010668f6.

Thanks for the contribution @izaakschroeder and again sorry for how long it has taken me to come back to this.

**Comment by izaakschroeder** [comment]
Thanks for getting it merged 😄

**Pull request closed without merging**